packages/fabrix/src/registry.tsx

```diff
@@ -24,153 +31,260 @@ export type Field = {
   type: FieldType;
 };
 
+export type CustomProps<P> = {
+  customProps: P;
+};
+
+type FieldLikeComponentProps<V = unknown> = BaseComponentProps<V> & {
+  path: string[];
+  subFields: Array<Field>;
+};
+
 /**
  * The component props that field renderer should implement.
  */
-export type FieldComponentProps<P extends UserProps = UserProps> =
+export type FieldComponentProps<UP extends UserProps = UserProps> =
+  FieldLikeComponentProps & CustomRendererProps<UP>;
+
+export type FieldsComponentProps<P = unknown> = CustomProps<P> & {
+  name: string;
+  className?: string;
+  value: Record<string, unknown>;
+};
+
+/**
+ * The component props that form field renderer should implement.
+ */
+export type FormFieldComponentProps<UP extends UserProps = UserProps> =
   BaseComponentProps &
-    CustomRendererProps<P> & {
-      path: string[];
-      subFields: Array<Field>;
+    CustomRendererProps<UP> & {
+      isRequired: boolean;
     };
 
 /**
- * The component props that table renderer should implement.
+ * The component props that form renderer should implement.
  */
-export type TableComponentProps<P extends UserProps = UserProps> =
-  CustomRendererProps<P> & {
-    name?: string;
-    className?: string;
-    headers: TableComponentHeader[];
-    values: Record<string, unknown>[];
-  };
+export type FormComponentProps<P = unknown> = CustomProps<P> & {
+  name: string;
+  className?: string;
+  renderFields: () => React.ReactNode;
+  renderSubmit: (
+    renderer: (props: {
+      submit: () => void;
+      isSubmitting: boolean;
+    }) => React.ReactElement,
+  ) => React.ReactNode;
+  renderReset: (
+    renderer: (props: { reset: () => void }) => React.ReactNode,
+  ) => React.ReactNode;
+};
+
 export type TableComponentHeader = Field & {
   render: ((rowValue: Record<string, unknown>) => React.ReactElement) | null;
```

Comment:
Is there a problem with ReactNode?
I have changed from ReactElement to ReactNode because I wanted to register a renderer that could return null before.
```suggestion
  render: ((rowValue: Record<string, unknown>) => React.ReactNode) | null;
```

Reply:
No. This might be unexpectedly degraded by me 🙏